compiler/noirc_frontend/src/tests/unused_items.rs

```diff
@@ -332,12 +332,13 @@ fn considers_struct_as_constructed_if_trait_method_is_called() {
 #[test]
 fn considers_struct_as_constructed_if_mentioned_in_let_type() {
     let src = "
-    struct Bar {}
+    pub struct Bar {}
 
-    pub fn main() {
-        let array = [];
+    pub fn foo(array: [Bar; 1]) {
```

Comment:
@asterite will adding `Bar` in the parameter here override that we are testing whether the struct is constructed in the let type?

Reply:
Good question! No, `Bar` as an input argument won't be considered as constructed. It's only the case in "return" positions (a return type, a let type, etc.). That said, it's true that if that were the case then this test wouldn't be testing this thing properly. I'll add another test 👍 , just a sec...

Reply:
Ah, actually I added `pub` here so that already breaks this...